minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java

```diff
@@ -196,7 +209,53 @@ protected abstract void startWorker(TachyonConf conf) throws IOException,
    * @param conf configuration of this test
    * @throws IOException when creating or deleting dirs failed
    */
-  protected abstract void setupTest(TachyonConf conf) throws IOException;
+  protected void setupTest(TachyonConf conf) throws IOException {
+    String tachyonHome = conf.get(Constants.TACHYON_HOME);
+    // Delete the tachyon home dir for this test from ufs to avoid permission problems
+    UnderFileSystemUtils.deleteDir(tachyonHome, conf);
+
+    // Create ufs dir. This must be called before starting UFS with UnderFileSystemCluster.get().
+    UnderFileSystemUtils.mkdirIfNotExists(conf.get(Constants.UNDERFS_ADDRESS), conf);
+
+    // Create storage dirs for worker
+    int numLevel = conf.getInt(Constants.WORKER_TIERED_STORE_LEVELS);
+    for (int level = 0; level < numLevel; level ++) {
+      String tierLevelDirPath =
+          String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level);
+      String[] dirPaths = conf.get(tierLevelDirPath).split(",");
+      for (String dirPath : dirPaths) {
+        UnderFileSystemUtils.mkdirIfNotExists(dirPath, conf);
+      }
+    }
+
+    // Start the UFS for integration tests. If this is for HDFS profiles, it starts miniDFSCluster
+    // (see also {@link tachyon.LocalMiniDFSCluster} and sets up the folder like
+    // "hdfs://xxx:xxx/tachyon*".
+    mUfsCluster = UnderFileSystemCluster.get(mTachyonHome, conf);
+
+    // Set the journal folder
+    String journalFolder =
+        mUfsCluster.getUnderFilesystemAddress() + "/journal" + RANDOM_GENERATOR.nextLong();
+    conf.set(Constants.MASTER_JOURNAL_FOLDER, journalFolder);
+
+    // Format the journal
+    UnderFileSystemUtils.mkdirIfNotExists(journalFolder, conf);
+    for (String masterServiceName : mMasterServiceNames) {
+      UnderFileSystemUtils.mkdirIfNotExists(PathUtils.concatPath(journalFolder, masterServiceName),
+          conf);
+    }
+    UnderFileSystemUtils.touch(journalFolder + "/_format_" + System.currentTimeMillis(), conf);
+
+    // If we are using the LocalMiniDFSCluster, we need to update the UNDERFS_ADDRESS to point to
+    // the cluster's current address. This must happen after UFS is started with
+    // UnderFileSystemCluster.get().
+    // TODO(andrew): Move logic to the integration-tests project so that we can use instanceof here
+    // instead of comparing classnames.
+    if (mUfsCluster.getClass().getSimpleName().equals("LocalMiniDFSCluster")) {
```

Comment:
I just moved this code from somewhere else, so I don't understand the comment either. Maybe @aaudiber can explain the comment?